crates/holochain_types/src/app.rs

```diff
@@ -235,8 +242,55 @@ impl automap::AutoMapped for InstalledApp {
     }
 }
 
+/// An installed App which is in the deactivated state.
+/// Note that `InstalledApp` should probably be renamed "ActiveApp" to show
+/// the distinction to this type, but didn't do that renaming yet.
+#[derive(
+    Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize, shrinkwraprs::Shrinkwrap,
+)]
+pub struct DeactivatedApp {
+    #[shrinkwrap(main_field)]
+    app: InstalledApp,
+    reason: DeactivationReason,
+}
+
+impl DeactivatedApp {
+    /// Constructor
+    pub fn new(app: InstalledApp, reason: DeactivationReason) -> Self {
+        Self { app, reason }
+    }
+}
+
+impl automap::AutoMapped for DeactivatedApp {
+    type Key = InstalledAppId;
+
+    fn key(&self) -> &Self::Key {
+        &self.app.installed_app_id
+    }
+}
+
+impl From<DeactivatedApp> for InstalledApp {
+    fn from(d: DeactivatedApp) -> Self {
+        d.app
+    }
+}
+
+/// The possible reasons for an app being deactivated
+#[derive(Clone, Debug, PartialEq, Eq, serde::Serialize, serde::Deserialize)]
```

Comment:
Currently this enum serializes like this over on the JS side:
```
{
          installed_app_id: 'app',
          cell_data: [ { cell_id: [Array], cell_nick: 'thedna' } ],
          status: { Inactive: { NeverActivated: null } }
        }
```
Can we have it be more ergonomic?

Reply:
This is addressed in the downstream PR https://github.com/holochain/holochain/pull/751